samcli/lib/intrinsic_resolver/intrinsics_symbol_table.py

```diff
@@ -0,0 +1,424 @@
+        resolver = self.default_type_resolver.get(resource_type, {}).get(resource_attribute) if resource_type else {}
+        if resolver:
+            if callable(resolver):
+                return resolver(logical_id, resource_attribute)
+            return resolver
+
+        attribute_resolver = self.common_attribute_resolver.get(resource_attribute, {})
+        if attribute_resolver:
+            if callable(attribute_resolver):
+                return attribute_resolver(logical_id)
+            return attribute_resolver
+
+        if ignore_errors:
+            return "${}".format(logical_id + "." + resource_attribute)
+        raise InvalidSymbolException(
+            "The {} is not supported in the logical_id_translator, default_type_resolver, or the attribute_resolver."
+            " It is also not a supported pseudo function".format(logical_id + "." + resource_attribute))
+
+    def verify_valid_fn_get_attribute(self, logical_id, resource_type):
+        """"
+        This function uses a CloudFormationResourceSpecification.json, which contains the list of attributes that can
+        be allowed in an Fn::Get.
+
+        This is currently not used because of filepath issues
+
+        Parameters
+        -----------
+        logical_id: str
+            This is the logical_id of the requested intrinsic's property in question.
+        resource_type: str
+            This is the resource_type of the requested intrinsic's property in question.
+
+        Returns
+        --------
+            A boolean that verifies
+        """
+        with open('CloudFormationResourceSpecification.json') as json_data:
+            resource_specification = json.load(json_data)
+            resource = self.resources.get(logical_id, {})
+
+            return resource_type in resource_specification.get(resource.get("Type", ""), {}).get("Attributes", {})
+
+    def arn_resolver(self, logical_id, service_name="lambda"):
+        """
+        This function resolves Arn in the format
+            arn:{partition_name}:{service_name}:{aws_region}:{account_id}:{function_name}
+
+        Parameters
+        -----------
+        logical_id: str
+            This the reference to the function name used
+        service_name: str
+            This is the service name used such as lambda or sns
+
+        Return
+        -------
+        The resolved Arn
+        """
+        aws_region = self.logical_id_translator.get(IntrinsicsSymbolTable.AWS_REGION) or self.handle_pseudo_region()
+        account_id = self.logical_id_translator.get(
+            IntrinsicsSymbolTable.AWS_ACCOUNT_ID) or self.handle_pseudo_account_id()
+        partition_name = self.handle_pseudo_partition()
+        function_name = logical_id
+        function_name = self.logical_id_translator.get(function_name) or function_name
+        str_format = "arn:{partition_name}:{service_name}:{aws_region}:{account_id}:{function_name}"
+        if service_name == "lambda":
```

Comment:
I wonder what other services have similar (custom) things in their arns.

Reply:
Can we look at a few other services? This function looks like it should be generic but has `function_name` in all arns, so it reads a little off. Also, I have the general concern that Lambda isn't the only special arn one here. Keeping this updated with every service, could be a big pain.

Reply:
The arn types are 
arn:partition:service:region:account-id:resource
arn:partition:service:region:account-id:resourcetype/resource
arn:partition:service:region:account-id:resourcetype/resource/qualifier
arn:partition:service:region:account-id:resourcetype/resource:qualifier
arn:partition:service:region:account-id:resourcetype:resource
arn:partition:service:region:account-id:resourcetype:resource:qualifier

The ones I looked at sns/sqs followed the default 
         str_format = "arn:{partition_name}:{service_name}:{aws_region}:{account_id}:{function_name}" 

Before, we also only supported dealing with Lambda Functions. This function will also only be called as a default case if they do not specify and override for resolving GetAtt for Arns.

Reply:
That's fine but then we shouldn't have the `str_format` be tied to `function_name`. My callout is an arn for sqs doesn't have a function_name, it has a queue_name. Since this is broad we need to make sure these reads as such. Reading it as it is written, makes it seems like lambda is only supported.